lib/season_statistics.rb

```diff
@@ -1,194 +1,141 @@
 require 'pry'
 
 class SeasonStatistics
-attr_reader :stat_tracker, :game_collection, :game_teams_collection, :teams_collection
+attr_reader :game_collection
 
   def initialize(game_collection, game_teams_collection, teams_collection)
     @game_collection = game_collection
     @game_teams_collection = game_teams_collection
     @teams_collection = teams_collection
   end
 
-  def current_season_games(season)
-    current_games = @game_collection.map do |game|
-     if game.season == season
-       game.game_id
-     end
-   end
-  current_games.compact
+#returns array of game ids for given season
+#try using find all and then map as opposed to compact
+  def current_season_game_ids(season)
+    season_game_ids = @game_collection.map do |game|
+      game.game_id if game.season == season
+    end
+    season_game_ids.compact
   end
 
-  def teams_hash(season)
-    teams = Hash.new(0)
-    season_teams = current_season_game_teams(season).map do |game|
-      game.team_id
+
+  #returns an array of game_team objects in given season
+    def current_season_game_teams(season)
```

Comment:
I think this is where our performance issue is coming from. We call this method about 7 times per season, and each time is iterates over the game_teams_collection. We will have to come back to this.